trillian/migrillian/core/trillian.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/google/certificate-transparency-go/scanner"
+	"github.com/google/trillian"
+	"github.com/google/trillian/client"
+	"github.com/google/trillian/crypto"
+	"github.com/google/trillian/types"
+)
+
+// TrillianTreeClient is a means of communicating with a Trillian pre-ordered
+// log tree.
+type TrillianTreeClient struct {
```

Comment:
Name could be improved. If it's specific to preordered log the name is too generic. PreorderedLogClient? Though it's not directly analagous to TrillianLogClient, which means perhaps Client shouldn't be there.

Reply:
Couldn't come up with anything better than `PreorderedLogClient`. Is that okay? Other ideas were: `(PO|Preordered)Log(Client|Instance|Gate|Shim|Thing)`.